node/src/proxy_server/mod.rs

```diff
@@ -1095,24 +1094,18 @@ impl IBCDHelper for IBCDHelperReal {
                 .dns_failure_retries
                 .insert(stream_key, dns_failure_retry);
         }
-
-        let tth_args = TryTransmitToHopperArgs {
-            main_cryptde: proxy.main_cryptde,
+        let retire_stream_key_sub_opt = if retire_stream_key {
```

Comment:
Maybe it would be better to move this logic into the `TryTransmitToHopperArgs::new()` constructor and pass in just the boolean flag?